stonesoup/predictor/kalman.py

```diff
@@ -413,8 +442,20 @@ class SqrtKalmanPredictor(KalmanPredictor):
         doc="A switch to do the prediction via a QR decomposition, rather than using a Cholesky "
             "decomposition.")
 
-    # This predictor returns a square root form of the Gaussian state prediction
-    _prediction_class = SqrtGaussianStatePrediction
+    # # This predictor returns a square root form of the Gaussian state prediction
+    # _prediction_class = SqrtGaussianStatePrediction
+
+    def _get_prediction(self, prior, state_vector, covar, timestamp):
+
+        # Get prediction type
+        pred_class = self._prediction_class(prior)
+        kwargs = {prop: getattr(prior, prop)
+                  for prop in pred_class.__dict__['_properties'].keys()}
+        kwargs['state_vector'] = state_vector
+        kwargs['sqrt_covar'] = covar
+        kwargs['timestamp'] = timestamp
+
+        return pred_class(**kwargs)
```

Comment:
This looks like repeated code. Could just exist in parent class?

Reply:
It was written like this because the `SqrtKalmanPredictor` requires the keyword argument _sqrt_covar_ over the `KalmanPredictor` needing the argument _covar_.
If we used _sqrt_covar_ in `KalmanPredictor` we would get an unexpected keyword argument type error.

Reply:
Could just overload this method to replace 'covar' with 'sqrt_covar' via some string variable, then call the method via super.